static/js/search_suggestion.js

```diff
@@ -697,6 +696,11 @@ export function get_search_result(base_query, query) {
         get_has_filter_suggestions,
     ];
 
+    if (!page_params.is_web_public_visitor) {
+        // Insert at the 3rd position.
+        filterers.splice(2, 0, get_sent_by_me_suggestions);
+    }
```

Comment:
I think we should just make `get_sent_by_my_suggestions` return nothing if `page_params.is_web_public_visitor`; that'll feel a lot cleaner than this splicing logic, which is structurally fragile (in that if we adjust the things above we might forget to change this).